src/main/java/reiff/brickbreaker/Controller.java

```diff
@@ -22,6 +33,24 @@ public Controller(Ball ball, Paddle paddle, List<Brick> bricks, BrickBreakerComp
 
     }
 
+    public void handleKeyEvent(int keyCode) {
+        int paddleSpeed = paddle.getSpeed();
+
+        // Move paddle left or right
+        if (keyCode == KeyEvent.VK_LEFT) {
+            paddle.x = Math.max(0, paddle.x - paddleSpeed);
+        } else if (keyCode == KeyEvent.VK_RIGHT) {
+            paddle.x = Math.min(view.getWidth() - paddle.width, paddle.x + paddleSpeed);
```

Comment:
If you fall this method while training the network view.getWidth() is always 0. 

It never moves to the correct place.

Reply:
Thank you so much! now it runs but it seems that the scores are too high, and it does not make sense: Generation: 1
Top 10 Scores for Generation 1:
Network 1 Score: 9989
Network 2 Score: 9989
Network 3 Score: 9989
Network 4 Score: 9989
Network 5 Score: 9988
Network 6 Score: 9988
Network 7 Score: 9988
Network 8 Score: 9986
Network 9 Score: 6661
Network 10 Score: 6661
Generation: 2
Top 10 Scores for Generation 2:
Network 1 Score: 9989
Network 2 Score: 9989
Network 3 Score: 9989
Network 4 Score: 9989
Network 5 Score: 9988
Network 6 Score: 9988
Network 7 Score: 9988
Network 8 Score: 9988
Network 9 Score: 9988
Network 10 Score: 9984.  Also after running it a few times with the debugger I noticed  double[] input = new double[1]; this number is the same for a few rounds and so the .guess(), is guessing the same values over and over again, I thought it was supposed to be random and not consistent over the rounds